code/modules/clothing/glasses/glasses.dm

```diff
@@ -231,10 +231,83 @@
 /obj/item/clothing/glasses/sunglasses/noir
 	name = "noir sunglasses"
 	desc = "Somehow these seem even more out-of-date than normal sunglasses."
-	actions_types = list(/datum/action/item_action/noir)
+	actions_types = list(/datum/action/item_action/noir, /datum/action/item_action/monologue)
 	var/noir_mode = 0
 	color_view = MATRIX_GREYSCALE
 
+/obj/item/clothing/glasses/sunglasses/noir/ui_action_click(mob/user, actiontype)
+	if(actiontype == /datum/action/item_action/noir)
+		attack_self()
+	else if(actiontype == /datum/action/item_action/monologue)
+		monologue(user)
+
+/obj/item/clothing/glasses/sunglasses/noir/proc/monologue(mob/user)
+	if(!ishuman(user))
+		return
+	var/mob/living/carbon/human/H = user
+
+	var/vox = (H.get_species() == "Vox" || H.get_species() == "Vox Armalis")
+	var/message = null //for visible_message
+	var/monologue = null
+	var/area/A = get_area(H)
+
+	if(H.is_muzzled())
+		message = "<B>[src]</B> struggles against \the [H.wear_mask]. [H.gender == MALE ? "H" : "Sh"]e's not giving up so easily."
+	else if(istype(H.l_hand, /obj/item/weapon/grab) && H.l_hand:state >= 3)//won't runtime because it quits if the first check fails

```

Comment:
I am going to shank you for every single one of the colons that aren't corrected in this PR by the time of your next commit 

🦊🔪